### PR TITLE
INC-1039: source Renovate config from shared HMPPS repo

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/INC-1039

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -1,49 +1,39 @@
...
-  ],
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": ["github>ministryofjustice/hmpps-renovate-config:node"],
   "assigneesFromCodeOwners": true,
-  "timezone": "Europe/London",
-  "schedule": [
-    "after 9am every weekday",
-    "before 5pm every weekday"
-  ],
-  "enabledManagers": ["npm", "nvm", "dockerfile", "docker-compose", "helmv3", "helm-values", "circleci"],
```
Comment (neilmendum) [inline renovate.json:13]: We now inherit this from https://github.com/ministryofjustice/hmpps-renovate-config/blob/4acc8951af908562c507a76b68ead835193c6a31/jvm.json#L4

File: renovate.json [hunk 2]
```diff
@@ -1,49 +1,39 @@
...
-    ":disableDependencyDashboard",
-    ":preserveSemverRanges"
-  ],
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": ["github>ministryofjustice/hmpps-renovate-config:node"],
   "assigneesFromCodeOwners": true,
-  "timezone": "Europe/London",
-  "schedule": [
-    "after 9am every weekday",
-    "before 5pm every weekday"
```
Comment (neilmendum) [inline renovate.json:11]: We now inherit this from https://github.com/ministryofjustice/hmpps-renovate-config/blob/4acc8951af908562c507a76b68ead835193c6a31/base.json#L8-L12

File: renovate.json [hunk 3]
```diff
@@ -1,49 +1,39 @@
...
-  ],
-  "enabledManagers": ["npm", "nvm", "dockerfile", "docker-compose", "helmv3", "helm-values", "circleci"],
   "packageRules": [
+    {
+      "matchManagers": ["npm"],
+      "rangeStrategy": "bump",
+      "matchUpdateTypes": ["minor", "patch"],
+      "groupName": "all non major NPM dependencies",
+      "groupSlug": "all-npm-minor-patch",
+      "stabilityDays": 3
```
Comment (neilmendum) [inline renovate.json:12]: As per [Renovate docs](https://docs.renovatebot.com/configuration-options/#rangestrategy)

> bump = e.g. bump the range even if the new version satisfies the existing range, e.g. ^1.0.0 -> ^1.1.0

This should avoid us needing to manually bump minor/patch dependencies manually

Then have introduced `stabilityDays`

> Number of days required before a new release is considered stable.

and `groupName` - to group them together in a single PR once they are at least 3 days old - this should ensure we aren't continually swamped with PRs. I spent some time testing this config on https://github.com/ministryofjustice/renovate-typescript-poc/pulls but can see how we find it?

**Review by ushkarev — APPROVED** [review]

File: renovate.json [hunk 4]
```diff
@@ -1,49 +1,39 @@
 {
-  "extends": [
-    "config:base",
-    ":disableDependencyDashboard",
-    ":preserveSemverRanges"
```
Comment (neilmendum) [inline renovate.json:5]: See https://github.com/ministryofjustice/hmpps-incentives-ui/pull/372/files#r1071431462

File: renovate.json [hunk 5]
```diff
@@ -1,49 +1,39 @@
...
+      "rangeStrategy": "bump",
+      "matchUpdateTypes": ["minor", "patch"],
+      "groupName": "all non major NPM dependencies",
+      "groupSlug": "all-npm-minor-patch",
+      "stabilityDays": 3
+    },
+    {
+      "matchPackageNames": ["typescript", "govuk-frontend"],
+      "rangeStrategy": "bump",
+      "stabilityDays": 0
```
Comment (neilmendum) [inline renovate.json:17]: Set this to 0 so we bump these straight away, as we have the `security` pipeline job `npm_outdated`

File: renovate.json [hunk 6]
```diff
@@ -1,27 +1,27 @@
...
+    },
     {
       "matchManagers": ["npm"],
       "matchPackageNames": ["@types/node"],
       "matchUpdateTypes": ["major"],
       "enabled": false
     },
-    {
-      "matchDatasources": ["docker"],
-      "allowedVersions": "18.12-bullseye-slim"
```
Comment (neilmendum) [inline renovate.json:23]: This is now pinned in https://github.com/ministryofjustice/hmpps-renovate-config/blob/4acc8951af908562c507a76b68ead835193c6a31/jvm.json#L7-L8

File: renovate.json [hunk 7]
```diff
@@ -36,14 +36,9 @@
...
-      "rangeStrategy": "bump"
-    },
-    {
-      "matchPackageNames": ["govuk-frontend"],
-      "rangeStrategy": "bump"
     }
-  ]
+  ],
+  "vulnerabilityAlerts": {
+    "stabilityDays": 0
```
Comment (neilmendum) [inline renovate.json:42]: As per [Renovate docs](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts)

> Renovate can read GitHub's Vulnerability Alerts to customize its Pull Requests

So configure Renovate to raise PRs for these straight away

**Review by ushkarev — APPROVED** [review]

**Pull request merged**